File: leetcode/hash-table/3761.cpp
class Solution {
    
public:

    int get_reverse(int num) {
        string s = to_string(num);
        reverse(s.begin(), s.end());
        return stoi(s);
    }

    int minMirrorPairDistance(vector<int>& nums) {
        int n = nums.size();
        vector<int> nums_r;
        for (auto num : nums) {
            nums_r.push_back(get_reverse(num));
        }

        unordered_map<int,int> d;
        int res = n;

        for (int i = 0; i < n; i++) {
            if (d.count(nums[i]))
                res = min(res, i - d[nums[i]]);
            d[nums_r[i]] = i;
        }

        return res == n ? -1 : res; 
    }
};

File: leetcode/hash-table/3761.py
class Solution:
    def minMirrorPairDistance(self, nums: List[int]) -> int:
        
        n = len(nums)
        nums_r = [int(str(num)[::-1]) for num in nums]
        d = dict()
        res = n

        for i in range(n):
            if nums[i] in d:
                res = min(res, i - d[nums[i]])
            d[nums_r[i]] = i

        # print(nums_r)
        # print(d)

        return res if res < n else -1
